frontend/src/components/RideCard.tsx

```diff
@@ -249,6 +264,31 @@ function DriverDisplay({
   );
 }
 
+function CompleteRideButton({
+  rideId,
+  userId,
+  driverId,
+}: {
+  rideId: string;
+  userId: string;
+  driverId: string | undefined;
+}) {
+  return (
+    <Button
+      disabled={driverId == undefined || userId !== driverId}
```

Comment:
Could we not render this button at all if they aren't the driver? I feel that having a button that they can never use may just be confusing. If you want to show the button then maybe a tooltip saying that the driver is the only one that can complete the ride would be good?

Reply:
Yeah that makes sense. Now only the driver can see the "complete" button